src/Time/Series.hs

```diff
@@ -174,3 +181,79 @@ instance ( KnownRatName unit
 -}
 unitsF :: forall unit . KnownRatName unit => Time unit -> String
 unitsF = seriesF @AllTimes
+
+{- | Class for time parsing.
+
+__Examples__
+
+>>> seriesP @'[Day, Hour, Minute, Second] @Minute "2d18h40m"
+Just (4000m)
+
+>>> seriesP @'[Day, Minute, Second] @Minute "2d1120m"
+Just (4000m)
+
+>>> seriesP @'[Hour, Minute, Second] @Second "1h1s"
+Just (3601s)
+
+>>> seriesP @'[Hour, Second, Millisecond] @Minute "90s"
+Just (1+1/2m)
+
+>>> seriesP @'[Hour, Second] @Second "11ns"
+Nothing
+
+__Note:__ The received list should be in descending order. It would be verified at compile-time.
+
+-}
+class SeriesP (units :: [Rat]) where
+    seriesP :: forall (someUnit :: Rat) . KnownRatName someUnit
+            => String -> Maybe (Time someUnit)
+
+instance SeriesP '[] where
+    seriesP _ = Nothing
+
+instance (KnownRatName unit) => SeriesP '[unit] where
+    seriesP :: forall (someUnit :: Rat) . KnownRatName someUnit
+            => String -> Maybe (Time someUnit)
+    seriesP ""  = Just $ Time 0
```

Comment:
I think this case (parsing empty string) should be in documentation. It's a hard problem, how we want to handle empty string. But this is not important at the moment. What's important is not to have undocumented behavior.